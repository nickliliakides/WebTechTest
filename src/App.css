.App {
  text-align: center;
}

.App .editor {
  text-align: left;
}

.App-logo {
  height: 40vmin;
  pointer-events: none;
}

h1 {
  color: #0e4b16;
}

@media (prefers-reduced-motion: no-preference) {
  .App-logo {
    animation: App-logo-spin infinite 20s linear;
  }
}

.App-header {
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.App-link {
  color: #61dafb;
}

.MuiButton-contained.Mui-disabled {
  color: 'white' !important;
  background-color: 'green' !important;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }

  to {
    transform: rotate(360deg);
  }
}

@media only screen and (max-width: 959px) {
  .edit_container {
    width: 100%;
  }

  .editor .MuiFormControl-root {
    width: 65vw !important;
  }
}