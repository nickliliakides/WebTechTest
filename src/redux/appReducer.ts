import { createStore, combineReducers, applyMiddleware } from 'redux';

import createSagaMiddleware from 'redux-saga';

import rootSaga from './sagas/rootSaga';

import * as sportwebservice from './actions/questionwebservice';

import { Question } from '../types/Question';

export interface QuizState {
  questions: Question[];
  selectedQuestion: Question | {};
}

const initialQuizState: QuizState = {
  questions: [],
  selectedQuestion: {},
};

function questionReduxReducer(
  state = initialQuizState,
  action: { type: string; payload: any }
) {
  const { type, payload } = action;

  switch (type) {
    case sportwebservice.Types.GET_QUESTIONS_SUCCESS:
      return {
        ...state,
        questions: [...payload.results, ...state.questions],
      };
    case sportwebservice.Types.DELETE_QUESTION_REQUEST:
      return {
        ...state,
        questions: [
          ...state.questions.filter((question) => question.id !== payload),
        ],
      };
    case sportwebservice.Types.SET_SELECTED_QUESTION:
      return {
        ...state,
        selectedQuestion: payload,
      };
    case sportwebservice.Types.UPDATE_SELECTED_QUESTION:
      return {
        questions: [
          payload,
          ...state.questions.filter((question) => question.id !== payload.id),
        ],
        selectedQuestion: {},
      };
    default:
      return state;
  }
}

const overallReducer = combineReducers({
  quiz: questionReduxReducer,
});

// Sage Middleware is used for fetching external resources.
const sagaMiddleware = createSagaMiddleware();

const store = createStore(overallReducer, applyMiddleware(sagaMiddleware));

sagaMiddleware.run(rootSaga);

export { store };
