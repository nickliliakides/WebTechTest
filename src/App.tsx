import React from 'react';
import './App.css';
import { Provider } from 'react-redux';
import { store } from './redux/appReducer';
import { ToastContainer } from 'react-toastify';
import 'react-toastify/dist/ReactToastify.css';
import { Grid } from '@material-ui/core';
import QuestionsList from './components/QuestionList';
import QuestionEditor from './components/QuestionEditor';

const App = () => {
  return (
    <Provider store={store}>
      <div style={{ padding: 25 }} className='App'>
        <ToastContainer
          position='bottom-center'
          autoClose={5000}
          hideProgressBar={false}
          newestOnTop
          closeOnClick
          rtl={false}
          pauseOnFocusLoss
          draggable
          pauseOnHover
        />
        <Grid container spacing={1}>
          <QuestionEditor />
          <QuestionsList />
        </Grid>
      </div>
    </Provider>
  );
};

export default App;
