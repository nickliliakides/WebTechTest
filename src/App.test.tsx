import React from 'react';
import ReactDOM from 'react-dom';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import App from './App';
import renderer from 'react-test-renderer';
import { Store } from 'redux';
const mockStore = configureStore([]);

describe('App component tests', () => {
  let store: Store;
  let element: JSX.Element;

  beforeEach(() => {
    store = mockStore({});

    element = (
      <Provider store={store}>
        <App />
      </Provider>
    );
  });

  it('Snapshot Testing', () => {
    const tree = renderer.create(element).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Expect to render App component', () => {
    const div = document.createElement('div');
    ReactDOM.render(element, div);
  });
});
