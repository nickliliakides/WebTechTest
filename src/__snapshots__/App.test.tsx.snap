// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component tests Snapshot Testing 1`] = `
<div
  className="App"
  style={
    Object {
      "padding": 25,
    }
  }
>
  <div
    className="Toastify"
  />
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1"
  >
    <div
      className="MuiGrid-root edit_container MuiGrid-item MuiGrid-grid-md-7"
      data-testid="editor"
    >
      <div
        className="editor"
      >
        <h1
          className="makeStyles-main_heading-2"
        >
          Edit Question
        </h1>
        <form
          className="makeStyles-root-1"
          onSubmit={[Function]}
        >
          <div
            className="react-reveal"
            style={
              Object {
                "opacity": undefined,
              }
            }
          >
            <div
              className="MuiPaper-root makeStyles-paper-4 MuiPaper-elevation1 MuiPaper-rounded"
            >
              <div
                className="makeStyles-paper_inner-5"
              >
                <h2
                  className="makeStyles-headings-3"
                  data-testid="header"
                >
                  Question
                </h2>
                <div>
                  <div
                    className="MuiFormControl-root MuiTextField-root"
                    data-testid="question-field"
                  >
                    <label
                      className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-disabled Mui-disabled"
                      data-shrink={false}
                      htmlFor="question"
                      id="question-label"
                    >
                      Question
                    </label>
                    <div
                      className="MuiInputBase-root MuiOutlinedInput-root Mui-disabled Mui-disabled MuiInputBase-formControl"
                      onClick={[Function]}
                    >
                      <input
                        aria-invalid={false}
                        autoFocus={false}
                        className="MuiInputBase-input MuiOutlinedInput-input Mui-disabled Mui-disabled"
                        disabled={true}
                        id="question"
                        name="question"
                        onAnimationStart={[Function]}
                        onBlur={[Function]}
                        onChange={[Function]}
                        onFocus={[Function]}
                        required={false}
                        type="text"
                        value=""
                      />
                      <fieldset
                        aria-hidden={true}
                        className="PrivateNotchedOutline-root-14 MuiOutlinedInput-notchedOutline"
                      >
                        <legend
                          className="PrivateNotchedOutline-legendLabelled-16"
                        >
                          <span>
                            Question
                          </span>
                        </legend>
                      </fieldset>
                    </div>
                  </div>
                </div>
                <div
                  className="makeStyles-flex-6"
                >
                  <h2
                    className="makeStyles-headings-3 makeStyles-left-7"
                  >
                    Answers
                  </h2>
                  <h2
                    className="makeStyles-headings-3 makeStyles-right-8"
                  >
                    Correct
                  </h2>
                </div>
                <div
                  className="makeStyles-flex-6"
                >
                  <div>
                    <div
                      className="MuiFormControl-root MuiTextField-root makeStyles-left-7"
                    >
                      <label
                        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-disabled Mui-disabled"
                        data-shrink={false}
                      >
                        Answer
                      </label>
                      <div
                        className="MuiInputBase-root MuiOutlinedInput-root Mui-disabled Mui-disabled MuiInputBase-formControl"
                        onClick={[Function]}
                      >
                        <input
                          aria-invalid={false}
                          autoFocus={false}
                          className="MuiInputBase-input MuiOutlinedInput-input Mui-disabled Mui-disabled"
                          disabled={true}
                          name="answer1"
                          onAnimationStart={[Function]}
                          onBlur={[Function]}
                          onChange={[Function]}
                          onFocus={[Function]}
                          required={false}
                          type="text"
                          value=""
                        />
                        <fieldset
                          aria-hidden={true}
                          className="PrivateNotchedOutline-root-14 MuiOutlinedInput-notchedOutline"
                        >
                          <legend
                            className="PrivateNotchedOutline-legendLabelled-16"
                          >
                            <span>
                              Answer
                            </span>
                          </legend>
                        </fieldset>
                      </div>
                    </div>
                  </div>
                  <label
                    className="MuiFormControlLabel-root makeStyles-right-8 Mui-disabled"
                  >
                    <span
                      aria-disabled={true}
                      className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-20 MuiCheckbox-root Component-root-18 PrivateSwitchBase-checked-21 Mui-checked Component-checked-19 PrivateSwitchBase-disabled-22 Mui-disabled Mui-disabled Mui-disabled"
                      data-testid="checkbox"
                      onBlur={[Function]}
                      onDragLeave={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      onKeyUp={[Function]}
                      onMouseDown={[Function]}
                      onMouseLeave={[Function]}
                      onMouseUp={[Function]}
                      onTouchEnd={[Function]}
                      onTouchMove={[Function]}
                      onTouchStart={[Function]}
                      tabIndex={-1}
                    >
                      <span
                        className="MuiIconButton-label"
                      >
                        <input
                          checked={true}
                          className="PrivateSwitchBase-input-23"
                          data-indeterminate={false}
                          disabled={true}
                          name="checked1"
                          onChange={[Function]}
                          type="checkbox"
                        />
                        <svg
                          aria-hidden={true}
                          className="MuiSvgIcon-root"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
                          />
                        </svg>
                      </span>
                    </span>
                    <span
                      className="MuiTypography-root MuiFormControlLabel-label Mui-disabled MuiTypography-body1"
                    >
                      
                    </span>
                  </label>
                </div>
                <div
                  className="makeStyles-flex-6"
                >
                  <div>
                    <div
                      className="MuiFormControl-root MuiTextField-root makeStyles-left-7"
                    >
                      <label
                        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-disabled Mui-disabled"
                        data-shrink={false}
                      >
                        Answer
                      </label>
                      <div
                        className="MuiInputBase-root MuiOutlinedInput-root Mui-disabled Mui-disabled MuiInputBase-formControl"
                        onClick={[Function]}
                      >
                        <input
                          aria-invalid={false}
                          autoFocus={false}
                          className="MuiInputBase-input MuiOutlinedInput-input Mui-disabled Mui-disabled"
                          disabled={true}
                          name="answer2"
                          onAnimationStart={[Function]}
                          onBlur={[Function]}
                          onChange={[Function]}
                          onFocus={[Function]}
                          required={false}
                          type="text"
                          value=""
                        />
                        <fieldset
                          aria-hidden={true}
                          className="PrivateNotchedOutline-root-14 MuiOutlinedInput-notchedOutline"
                        >
                          <legend
                            className="PrivateNotchedOutline-legendLabelled-16"
                          >
                            <span>
                              Answer
                            </span>
                          </legend>
                        </fieldset>
                      </div>
                    </div>
                  </div>
                  <label
                    className="MuiFormControlLabel-root makeStyles-right-8 Mui-disabled"
                  >
                    <span
                      aria-disabled={true}
                      className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-20 MuiCheckbox-root Component-root-18 PrivateSwitchBase-disabled-22 Mui-disabled Mui-disabled Mui-disabled"
                      onBlur={[Function]}
                      onDragLeave={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      onKeyUp={[Function]}
                      onMouseDown={[Function]}
                      onMouseLeave={[Function]}
                      onMouseUp={[Function]}
                      onTouchEnd={[Function]}
                      onTouchMove={[Function]}
                      onTouchStart={[Function]}
                      tabIndex={-1}
                    >
                      <span
                        className="MuiIconButton-label"
                      >
                        <input
                          checked={false}
                          className="PrivateSwitchBase-input-23"
                          data-indeterminate={false}
                          disabled={true}
                          name="checked2"
                          onChange={[Function]}
                          type="checkbox"
                        />
                        <svg
                          aria-hidden={true}
                          className="MuiSvgIcon-root"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                          />
                        </svg>
                      </span>
                    </span>
                    <span
                      className="MuiTypography-root MuiFormControlLabel-label Mui-disabled MuiTypography-body1"
                    >
                      
                    </span>
                  </label>
                </div>
                <div>
                  <button
                    className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-save_btn-10 MuiButton-containedPrimary Mui-disabled Mui-disabled"
                    data-testid="save-button"
                    disabled={true}
                    onBlur={[Function]}
                    onDragLeave={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    onKeyUp={[Function]}
                    onMouseDown={[Function]}
                    onMouseLeave={[Function]}
                    onMouseUp={[Function]}
                    onTouchEnd={[Function]}
                    onTouchMove={[Function]}
                    onTouchStart={[Function]}
                    tabIndex={-1}
                    type="submit"
                  >
                    <span
                      className="MuiButton-label"
                    >
                      <span
                        className="MuiButton-startIcon MuiButton-iconSizeMedium"
                      >
                        <svg
                          aria-hidden={true}
                          className="MuiSvgIcon-root"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M17 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V7l-4-4zm-5 16c-1.66 0-3-1.34-3-3s1.34-3 3-3 3 1.34 3 3-1.34 3-3 3zm3-10H5V5h10v4z"
                          />
                        </svg>
                      </span>
                      Save
                    </span>
                  </button>
                  <button
                    className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-add_btn-11 MuiButton-containedPrimary Mui-disabled Mui-disabled"
                    data-testid="add-button"
                    disabled={true}
                    onBlur={[Function]}
                    onClick={[Function]}
                    onDragLeave={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    onKeyUp={[Function]}
                    onMouseDown={[Function]}
                    onMouseLeave={[Function]}
                    onMouseUp={[Function]}
                    onTouchEnd={[Function]}
                    onTouchMove={[Function]}
                    onTouchStart={[Function]}
                    tabIndex={-1}
                    type="button"
                  >
                    <span
                      className="MuiButton-label"
                    >
                      <span
                        className="MuiButton-startIcon MuiButton-iconSizeMedium"
                      >
                        <svg
                          aria-hidden={true}
                          className="MuiSvgIcon-root"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm-2 10h-4v4h-2v-4H7v-2h4V7h2v4h4v2z"
                          />
                        </svg>
                      </span>
                      Add Answer
                    </span>
                  </button>
                  <button
                    className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-preview_btn-12 MuiButton-containedPrimary Mui-disabled Mui-disabled"
                    data-testid="preview-button"
                    disabled={true}
                    onBlur={[Function]}
                    onClick={[Function]}
                    onDragLeave={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    onKeyUp={[Function]}
                    onMouseDown={[Function]}
                    onMouseLeave={[Function]}
                    onMouseUp={[Function]}
                    onTouchEnd={[Function]}
                    onTouchMove={[Function]}
                    onTouchStart={[Function]}
                    tabIndex={-1}
                    type="button"
                  >
                    <span
                      className="MuiButton-label"
                    >
                      <span
                        className="MuiButton-startIcon MuiButton-iconSizeMedium"
                      >
                        <svg
                          aria-hidden={true}
                          className="MuiSvgIcon-root"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M12 4.5C7 4.5 2.73 7.61 1 12c1.73 4.39 6 7.5 11 7.5s9.27-3.11 11-7.5c-1.73-4.39-6-7.5-11-7.5zM12 17c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5zm0-8c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3z"
                          />
                        </svg>
                      </span>
                      Preview
                    </span>
                  </button>
                </div>
              </div>
            </div>
          </div>
        </form>
        <div />
      </div>
    </div>
    <div
      className="MuiCircularProgress-root makeStyles-loader-33 MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
      role="progressbar"
      style={
        Object {
          "height": 80,
          "width": 80,
        }
      }
    >
      <svg
        className="MuiCircularProgress-svg"
        viewBox="22 22 44 44"
      >
        <circle
          className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
          cx={44}
          cy={44}
          fill="none"
          r={19}
          strokeWidth={6}
          style={Object {}}
        />
      </svg>
    </div>
  </div>
</div>
`;
